pennylane/template.py

```diff
@@ -0,0 +1,225 @@
+
+    num_wires = len(wires)
+    for i in range(num_wires) if periodic else range(num_wires-1):
+        imprimitive_gate(wires=[wires[i], wires[(i+r) % num_wires]])
+
+
+def CVNeuralNet(weights, wires=None):
+    """pennylane.template.CVNeuralNet(weights, wires)
+    A CV Quantum Neural Network
+
+    Implements the CV Quantum Neural Network (CVQNN) architecture from
+    :cite:`killoran2018continuous` for an arbitrary number of wires
+    and layers.
+
+    See :func:`CVNeuralNetLayer` for details of the expected format of
+    input parameters.
+
+    Args:
+        weights (array[array]): Array of weights for each layer of the CV
+                                neural network
+        wires (Sequence[int]): Wires the CVQNN should act on
+    """
+    for layer_weights in weights:
+        CVNeuralNetLayer(*layer_weights, wires=wires)
+
+
+def CVNeuralNetLayer(theta1, phi1, s, theta2, phi2, r, k, wires=None): #pylint: disable-msg=too-many-arguments
+    """pennylane.template.CVNeuralNetLayer(theta1, phi1, s, theta2, phi2, r, k, wires)
+    A single layer of a CV Quantum Neural Network
+
+    Implements a single layer from the the CV Quantum Neural Network (CVQNN)
+    architecture of :cite:`killoran2018continuous` for an arbitrary number
+    of wires and layers.
+
+    .. note::
+
+       The CV neural network architecture includes :class:`~.Kerr` operations. Make sure to use a suitable device,
+       such as the :code:`strawberryfields.fock` device of the `PennyLane-SF <https://github.com/XanaduAI/pennylane-sf>`_ plugin.
+
+    Args:
+        theta1 (array[float]): length ``len(wires)*(len(wires)-1)/2`` array of transmittivity angles
+        phi1 (array[float]): length ``len(wires)*(len(wires)-1)/2`` array of phase angles
+        s (array[float]): length ``len(wires)`` arrays of squeezing amounts for :class:`~.Squeezing` operations
+        theta2 (array[float]): length ``len(wires)*(len(wires)-1)/2`` array of transmittivity angles
+        phi2 (array[float]): length ``len(wires)*(len(wires)-1)/2`` array of phase angles
+        r (array[float]): length ``len(wires)`` arrays of displacement magnitudes for :class:`~.Displacement` operations
+        k (array[float]): length ``len(wires)`` arrays of kerr parameters for :class:`~.Kerr` operations
+<<<<<<< HEAD
+=======
+        tolerance (int): The number of decimal places to use when determining whether a gate parameter obtained is so close
+                         to trivial that the gate is effectively an Identity and can be skipped.
+>>>>>>> ef7ee29cf4c06c091a9ae6d1c2d26f0ddf459c9c
+        wires (Sequence[int]): Wires the layer should act on
+    """
+    Interferometer(theta=theta1, phi=phi1, wires=wires)
+    for i, wire in enumerate(wires):
+        Squeezing(s[i], 0., wires=wire)
+
+    Interferometer(theta=theta2, phi=phi2, wires=wires)
+
+    for i, wire in enumerate(wires):
+        Displacement(r[i], 0., wires=wire)
```

Comment:
leaving a reminder here so we remember to update to complex displacement params

Reply:
Done.